include/z64cutscene_commands.h

```diff
@@ -11,445 +11,273 @@
+ * For commands that only need to last one frame, set `endFrame` to `startFrame + 1`.
  */
-#define CS_CMD_08(continueFlag, roll, frame, viewAngle, xPos, yPos, zPos, unused) \
-    CMD_BBH(continueFlag, roll, frame), CMD_F(viewAngle), CMD_HH(xPos, yPos), CMD_HH(zPos, unused)
+#define CS_MISC(cmd, startFrame, endFrame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7, unused8, unused9, unused10) \
+    CMD_HH(cmd, startFrame), CMD_HH(endFrame, unused0), \
+    CMD_W(unused1), CMD_W(unused2), CMD_W(unused3), CMD_W(unused4), CMD_W(unused5), \
+    CMD_W(unused6), CMD_W(unused7), CMD_W(unused8), CMD_W(unused9), CMD_W(unused10)
 
 /**
- * ARGS
- *   s32 entries (e)
- * FORMAT
- *   00000009 eeeeeeee
- *   size = 0x8
+ * Declares a list of `CS_LIGHT_SETTING` entries.
  */
-#define CS_CMD_09_LIST(entries) CS_CMD_09, CMD_W(entries)
+#define CS_LIGHT_SETTING_LIST(entries) \
+    CS_CMD_LIGHT_SETTING, CMD_W(entries)
 
 /**
- * ARGS
- *   s16 unk (u), s16 startFrame (s), s16 endFrame (e), s16 unk2 (v), s16 unk3 (w), s16 unk4 (x)
- * FORMAT
- *   Capital U is Unused
- *   uuuussss eeeevvww xxUUUUUU
- *   size = 0xC
+ * Changes the environment lights to the specified setting.
+ * The lighting change will take place immediately with no blending.
  */
-#define CS_CMD_09(unk, startFrame, endFrame, unk2, unk3, unk4, unused0, unused1) \
-    CMD_HH(unk, startFrame), CMD_HBB(endFrame, unk2, unk3), CMD_BBH(unk4, unused0, unused1)
+#define CS_LIGHT_SETTING(lightSetting, frame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7, unused8) \
+    CMD_HH((lightSetting + 1), frame), CMD_HH(unused0, unused1), \
+    CMD_W(unused2), CMD_W(unused3), CMD_W(unused4), CMD_W(unused5), CMD_W(unused6), \
+    CMD_W(unused7), CMD_W(unused8), 0x00000000, 0x00000000, 0x00000000
 
 /**
- * ARGS
- *   s32 cmdType (c), s32 entries (e)
- * FORMAT
- *   cccccccc eeeeeeee
- *   size = 0x8
+ * Declares a list of `CS_RUMBLE_CONTROLLER` entries.
  */
-#define CS_UNK_DATA_LIST(cmdType, entries) CMD_W(cmdType), CMD_W(entries)
+#define CS_RUMBLE_CONTROLLER_LIST(entries) \
+    CS_CMD_RUMBLE_CONTROLLER, CMD_W(entries)
 
-/**
- * ARGS
- *   s32 unk1 (a), s32 unk2 (b), s32 unk3 (c), s32 unk4 (d), s32 unk5 (e), s32 unk6 (f),
- *   s32 unk7 (g), s32 unk8 (h), s32 unk9 (i), s32 unk10 (j), s32 unk11 (k), s32 unk12 (l)
- * FORMAT
- *   aaaaaaaa bbbbbbbb cccccccc dddddddd eeeeeeee ffffffff gggggggg hhhhhhhh iiiiiiii jjjjjjjj kkkkkkkk llllllll
- *   size = 0x30
- */
-#define CS_UNK_DATA(unk1, unk2, unk3, unk4, unk5, unk6, unk7, unk8, unk9, unk10, unk11, unk12) \
-    CMD_W(unk1), CMD_W(unk2), CMD_W(unk3), CMD_W(unk4), CMD_W(unk5), CMD_W(unk6), \
-    CMD_W(unk7), CMD_W(unk8), CMD_W(unk9), CMD_W(unk10), CMD_W(unk11), CMD_W(unk12)
+#define CS_RUMBLE_CONTROLLER(unk, startFrame, endFrame, sourceStrength, duration, decreaseRate, unused0, unused1) \
+    CMD_HH(unk, startFrame), CMD_HBB(endFrame, sourceStrength, duration), CMD_BBH(decreaseRate, unused0, unused1)
```

Comment:
endFrame can be changed to unused (and in `CsCmdRumble`)

bonus points to this one for having "duration" and not using endFrame